app/script/entity/message/MemberMessage.js

```diff
@@ -98,16 +154,24 @@ z.entity.MemberMessage = class MemberMessage extends z.entity.SystemMessage {
           const senderJoined = this.otherUser().id === this.user().id;
           if (senderJoined) {
             const userJoinedStringId = this.user().is_me
-              ? z.string.conversationMemberJoinSelfYou
-              : z.string.conversationMemberJoinSelf;
-            return z.l10n.text(userJoinedStringId);
+              ? z.string.conversationMemberJoinedSelfYou
+              : z.string.conversationMemberJoinedSelf;
+            return z.l10n.text(userJoinedStringId, this.senderName());
           }
 
           const userJoinedStringId = this.user().is_me
-            ? z.string.conversationMemberJoinYou
-            : z.string.conversationMemberJoin;
-
-          return z.l10n.text(userJoinedStringId, this._generateNameString());
+            ? this.exceedsMaxVisibleUsers()
+              ? z.string.conversationMemberJoinedYouMore
+              : z.string.conversationMemberJoinedYou
+            : this.exceedsMaxVisibleUsers()
```

Comment:
Conjoined ternaries are hard to read. Can we pull these apart?

Reply:
We could put these in if-blocks, i will try if that makes it more readable.